### PR TITLE
De-flake TestReplicaCorruption.

### LINKED ISSUE #3632: Failed tests (10656): TestReplicaCorruption
The following test appears to have failed:

[#10656](https://circleci.com/gh/cockroachdb/cockroach/10656):

```
I0106 00:00:39.678314 532 gossip/client.go:161  sent 1 info(s) to node 3
I0106 00:00:39.678611 532 gossip/client.go:75  closing client to node 3 (127.0.0.1:40148): gossip/client.go:251: client closed
I0106 00:00:39.678902 532 gossip/client.go:75  closing client to node 1 (127.0.0.1:35743): gossip/client.go:251: client closed
I0106 00:00:39.679096 532 gossip/client.go:75  closing client to node 3 (127.0.0.1:40148): gossip/client.go:251: client closed
W0106 00:00:39.679297 532 rpc/server.go:309  rpc: write response failed: write tcp 127.0.0.1:40148->127.0.0.1:39056: use of closed network connection
--- FAIL: TestConvergence (0.46s)
	convergence_test.go:33: expected a fully-connected network within 20 cycles; took 21
FAIL
FAIL	github.com/cockroachdb/cockroach/gossip	0.671s
=== RUN   TestParseResolverSpec
--- PASS: TestParseResolverSpec (0.00s)
=== RUN   TestGetAddress
--- PASS: TestGetAddress (0.00s)
PASS
ok  	github.com/cockroachdb/cockroach/gossip/resolver	1.025s
?   	github.com/cockroachdb/cockroach/gossip/simulation	[no test files]
=== RUN   TestKeySorting
--- PASS: TestKeySorting (0.00s)
=== RUN   TestMakeKey
--
I0106 00:03:51.011080 1010 storage/replica.go:1567  gossiping sentinel from store 1, range 1
I0106 00:03:51.011407 1010 storage/replica.go:1573  gossiping first range from store 1, range 1
I0106 00:03:51.012471 1010 storage/replica.go:1112  reproposing 1 commands after empty entry
E0106 00:03:51.014406 1010 storage/replica.go:1206  error executing raft command: write too old: timestamp 0.000000000,12 <= 0.000000000,12
I0106 00:03:51.015680 1010 storage/engine/rocksdb.go:137  closing in-memory rocksdb instance
--- FAIL: TestReplicaCorruption (0.01s)
	replica_test.go:2960: unexpected error: %!s(<nil>)
=== RUN   TestChangeReplicasDuplicateError
I0106 00:03:51.018158 1010 raft/raft.go:446  [group 1] 1 became follower at term 5
I0106 00:03:51.018351 1010 raft/raft.go:234  [group 1] newRaft 1 [peers: [1], term: 5, commit: 10, applied: 10, lastindex: 10, lastterm: 5]
I0106 00:03:51.018448 1010 raft/raft.go:526  [group 1] 1 is starting a new election at term 5
I0106 00:03:51.018547 1010 raft/raft.go:459  [group 1] 1 became candidate at term 6
I0106 00:03:51.018642 1010 raft/raft.go:508  [group 1] 1 received vote from 1 at term 6
I0106 00:03:51.018757 1010 raft/raft.go:487  [group 1] 1 became leader at term 6
I0106 00:03:51.020471 1010 storage/replica.go:1558  gossiping cluster id test from store 1, range 1
I0106 00:03:51.023071 1010 storage/replica_command.go:1199  range 1: new leader lease replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC +1.000s
--
I0106 00:04:13.956042 1010 storage/engine/rocksdb.go:137  closing in-memory rocksdb instance
I0106 00:04:13.956380 1010 storage/engine/rocksdb.go:137  closing in-memory rocksdb instance
--- PASS: TestLeaderAfterSplit (0.25s)
=== RUN   Example_rebalancing
--- PASS: Example_rebalancing (0.71s)
FAIL
FAIL	github.com/cockroachdb/cockroach/storage	25.866s
=== RUN   TestBatchBasics
I0106 00:03:44.373312 990 storage/engine/rocksdb.go:137  closing in-memory rocksdb instance
--- PASS: TestBatchBasics (0.01s)
=== RUN   TestBatchGet
I0106 00:03:44.378013 990 storage/engine/rocksdb.go:137  closing in-memory rocksdb instance
--- PASS: TestBatchGet (0.01s)
=== RUN   TestBatchMerge
I0106 00:03:44.385242 990 storage/engine/rocksdb.go:137  closing in-memory rocksdb instance
--- PASS: TestBatchMerge (0.01s)
=== RUN   TestBatchProto

```
Please assign, take a look and update the issue accordingly.

### PR DESCRIPTION
**Pull request by bdarnell**
Fixes #3632

[Review on Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/3729)

### CODE REVIEW

**Comment by tschottdorf** [comment]
LGTM

File: storage/replica_test.go [hunk 1]
```diff
@@ -2946,30 +2946,34 @@ func TestAppliedIndex(t *testing.T) {
...
-	err = setAppliedIndex(tc.rng.store.Engine(), tc.rng.RangeID, newIndex)
-	if err != nil {
+	// First send a regular command.
+	args := putArgs(roachpb.Key("test1"), []byte("value"))
+	if _, err := client.SendWrapped(tc.Sender(), tc.rng.context(), &args); err != nil {
 		t.Fatal(err)
 	}
+
 	// Should mark replica corrupt (and panic as a result) since we messed
 	// with the applied index.
```
Comment (tamird) [inline storage/replica_test.go]: Comment outdated

Comment (bdarnell) [inline storage/replica_test.go]: Fixed

**Comment by tamird** [comment]
LGTM

**Pull request merged**